docs/Coding-Guidelines/ErrorHandlingAndLogging.md

```diff
@@ -0,0 +1,17 @@
+# Error Handling and Logging
+
+# Error Handling
+
+<span style="color:green">**DO**</span> Throw exceptions to report exceptional failures.
+
+<span style="color:green">**DO**</span> Use [WIL error handling helpers](https://github.com/Microsoft/wil/wiki/Error-handling-helpers) when handling error code failures (`THROW_*()`, `RETURN_*()`, `FAIL_FAST_*()`, `LOG_*()`).
+
+# Logging
+
+<span style="color:green">**DO**</span> Rely on [WIL error handling helpers](https://github.com/Microsoft/wil/wiki/Error-handling-helpers) to log failures.
+
+ <span style="color:green">**DO**</span> Use `TraceLoggingWrite()` to log non-failure information.
+
```

Comment:
```suggestion

<span style="color:green">**DON'T**</span> Define error 'HRESULTS' using FACILITY_ITF.  This is a general-purpose category that results in confusing logging messages.  Instead, use an existing HRESULT that approximates the error described. ```

Reply:
FACILITY_ITF is how you define new HRESULTs (if you're not in the Windows source tree).

You should use existing HRESULTs if possible, but if not then yes you should use FACILITY_ITF to define a new HRESULT.

How's this?

<span style="color:green">**DO**</span> Use existing 'HRESULTS' if you can. Be careful to pick an existing HRESULT whose symbolic name and message convey the intent and don't mislead the developer, administrator or user.

Reply:
that works